src/commands/registry.rs

```diff
@@ -143,6 +143,9 @@ impl Login {
             let mut buf = String::new();
             let mut stdin = std::io::stdin().lock();
             stdin.read_to_string(&mut buf)?;
+            // trim trailing newlines and carriage returns (see https://blog.v-gar.de/2019/04/rust-remove-trailing-newline-after-input)
+            let len = buf.trim_end_matches(&['\r', '\n'][..]).len();
+            buf.truncate(len);
```

Comment:
Is there a reason to use this length to truncate the original `buf` rather than return `buf.trim_end_matches(...).to_owned()`?  That feels more readable to me and I'm not worried if it incurs an extra allocation.  But I'm not familiar with this code and don't know the nuances of the string functions - I'll defer to you and Radu on this!

Reply:
I think the only justification for the length variant would be to "fix" the faulty `buf` by mutating it. But yeah I agree that yours is much more readable. Pushed a change 👍